views/products.hbs

```diff
@@ -0,0 +1,137 @@
+                        <p class="cart__items">0 items</p>
+                    </div>
+
+                </div>
+                <!-- Simulate a smartphone / tablet -->
+
+
+                <!-- Top Navigation Menu -->
+
+                <div class="topnav">
+                    <a href="javascript:void(0);" class="icon" onclick="myFunction()">
+                        <i class="fa fa-bars"></i>
+                    </a>
+                </div>
+
+                <!-- End smartphone / tablet look -->
+
+
+            </nav>
+        </div>
+    </header>
+
+    <section class="mobile-menu">
+        <div class="row">
+            <div id="myLinks">
+                <a href="#">Register</a>
+                <a href="#">Sign In</a>
+                <a href="#">Home</a>
+                <a href="#">Products</a>
+            </div>
+        </div>
+    </section>
+
+
+    <section class="section-products">
+        <div class="row">
+            <div class="section-product-categories">
+                <ul>
+                    {{#each categories}}
+                    <li><a href="/products?categoryID={{this.id}}">{{this.name}}</a></li>
+                    {{/each}}
+                </ul>
+            </div>
+            <div class=" section-all-products">
+                {{#each products}}
+                <div class="section-product">
+                    <p class="section-product-title">
+                        {{this.name}}
+                    </p>
+                    <div class="section-product-info">
+                        <figure class="section-product-image">
+                            <img src="{{this.imageURL}}" alt="{{this.name}}">
+                        </figure>
+                        <p class="section-product-details">
+                            {{this.description}}
+                        </p>
+                    </div>
+                    <div class="section-product-cost">
+                        <p class="section-product-rate">
+                            MRP Rs.{{this.price}}
+                        </p>
+                        <br>
+                        <form action="/cartValue" method="POST">
+                            <button class="BuyBtn">Buy Now</button>
+                        </form>
+                    </div>
+                </div>
+
+                {{/each}}
+            </div>
+        </div>
+    </section>
+
+    <section class="section-modal-box">
+        <div class="modal-cart-header">
+            <p>My Cart</p>
+            <p onclick="closeModalBox()">X</p>
+        </div>
+        <div class="modal-cart-items">
+
+        </div>
+
+        <button class="modal-cart-start-shopping">Start Shopping</button>
+    </section>
+
+
+    {{> footer}}
+
+</body>
+<script src="./res/js/cart.js"></script>
+
+<script>
+    let btns = document.getElementsByClassName('BuyBtn');
+
+    [].forEach.call(btns, each => {
+
+        each.addEventListener('click', (e) => {
```

Comment:
Instead of binding event on each button separately bind it on a containing element and inside event listener validate if this is the element where event needs to be binded. Read about event delegation.